### ISSUE TITLE
[Bug]: `prefer-read-only-props` rule doesn't work with namespaces

### ISSUE THREAD

**Issue by Mnigos** [jsx-eslint/eslint-plugin-react#3786]
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### Description Overview

`prefer-read-only-props` rule should work with namespace syntax

```ts
namespace ItemsListElementSkeleton {
  export interface Props {
    withArtists?: boolean
    withPlayedAt?: boolean
    position?: number
    positionSize?: ItemPosition.Props['size']
    positionClassName?: string
    withPlaysOrPlayTime?: boolean
  }
}

function ItemsListElementSkeleton({
  position,
  positionSize,
  positionClassName,
  withArtists,
  withPlayedAt,
  withPlaysOrPlayTime,
}: ItemsListElementSkeleton.Props) {
...
}

export { ItemsListElementSkeleton }
```

Should throw an error

```
Prop '...' should be read-only.
```

### Expected Behavior

Should throw error if props are not readonly

### eslint-plugin-react version

v7.35.0

### eslint version

v8.57

### node version

v20.10.0

**Comment by ljharb** [comment]
@Mnigos like everywhere on github, if there were an update, it'd be posted on the issue.
